xikolo-ios/ViewControllers/Courses/CourseListViewController.swift

```diff
@@ -46,11 +46,32 @@ class CourseListViewController : AbstractCourseListViewController {
                 let indexPath = collectionView!.indexPathForCell(cell)
                 let course = resultsController.objectAtIndexPath(indexPath!) as! Course
                 vc.course = course
+            case "ShowCourseDetails"?:
+                let vc = segue.destinationViewController as! CourseDetailViewController
+                let course = sender as! Course
+                vc.course = course
             default:
                 break
         }
     }
 
+    override func shouldPerformSegueWithIdentifier(identifier: String, sender: AnyObject?) -> Bool {
+        switch identifier {
+        case "ShowCourseDetailSegue":
+            let cell = sender as! CourseCell
+            let indexPath = collectionView!.indexPathForCell(cell)
+            let course = resultsController.objectAtIndexPath(indexPath!) as! Course
+            if course.is_enrolled {
+                return true
+            } else {
+                performSegueWithIdentifier("ShowCourseDetails", sender: course)
+                return false
+            }
```

Comment:
Please decide which segue to trigger before triggering. Don't trigger one and then decide here that you should have triggered a different one.